driver_test.go

```diff
@@ -12,26 +12,263 @@
+		if err != nil {
+			return nil, err
+		}
+	} else {
+		adminClient, err = adminapi.NewDatabaseAdminClient(ctx)
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	dataClient, err := spanner.NewClient(ctx, dsn)
+	if err != nil {
+		return nil, err
+	}
+
+	curs := &Connector{
+		ctx:         ctx,
+		client:      dataClient,
+		adminClient: adminClient,
+	}
+	return curs, nil
+}
+
+func (c *Connector) Close() {
+	c.client.Close()
+	c.adminClient.Close()
+}
+
+func init() {
+
+	var projectId, instanceId, databaseId string
+	var ok bool
+
+	// Get environment variables or set to default.
+	if instanceId, ok = os.LookupEnv("SPANNER_TEST_INSTANCE"); !ok {
+		instanceId = "test-instance"
+	}
+	if projectId, ok = os.LookupEnv("SPANNER_TEST_PROJECT"); !ok {
+		projectId = "test-project"
+	}
+	if databaseId, ok = os.LookupEnv("SPANNER_TEST_DBID"); !ok {
+		databaseId = "gotest"
+	}
+
+	// Derive data source name.
+	dsn = "projects/" + projectId + "/instances/" + instanceId + "/databases/" + databaseId
+}
+
+// Used to check if error contains expected string
+// If want is the empty string, no error is expected
+func ErrorContainsStr(err error, want string) bool {
+	if want == "" && err != nil {
+		return false
+	}
+	if err == nil {
+		return want == ""
+	}
+	return strings.Contains(err.Error(), want)
+}
+
+// Executes DDL statements.
+func executeDdlApi(curs *Connector, ddls []string) (err error) {
+
+	op, err := curs.adminClient.UpdateDatabaseDdl(curs.ctx, &adminpb.UpdateDatabaseDdlRequest{
+		Database:   dsn,
+		Statements: ddls,
+	})
+	if err != nil {
+		return err
+	}
+	if err := op.Wait(curs.ctx); err != nil {
+		return err
 	}
-	connector, err := driver.OpenConnector("projects/$PROJECT/instances/$INSTANCE/databases/$DATABASE")
+	return nil
+}
+
+// Executes DML using the client library.
+func ExecuteDMLClientLib(dml []string) (err error) {
+
+	// Open client/
+	ctx := context.Background()
+	client, err := spanner.NewClient(ctx, dsn)
+	if err != nil {
+		return err
+	}
+	defer client.Close()
+
+	// Put strings into spanner.Statement structure.
+	var stmts []spanner.Statement
+	for _, line := range dml {
+		stmts = append(stmts, spanner.NewStatement(line))
+	}
+
+	// Execute statements.
+	_, err = client.ReadWriteTransaction(ctx, func(ctx context.Context, txn *spanner.ReadWriteTransaction) error {
+		stmts := stmts
+		rowCounts, err := txn.BatchUpdate(ctx, stmts)
+		if err != nil {
+			return err
+		}
+		fmt.Printf("Executed %d SQL statements using Batch DML.\n", len(rowCounts))
+		return nil
+	})
+
+	return err
+}
+
+func TestQueryContext(t *testing.T) {
+
+	// Set up test table.
+	curs, err := NewConnector()
 	if err != nil {
 		log.Fatal(err)
 	}
-	db := sql.OpenDB(connector)
-	_ = db // Use db.
+	defer curs.Close()
+
+	err = executeDdlApi(curs, []string{`CREATE TABLE TestQueryContext (
+		A   STRING(1024),
+		B  STRING(1024),
+		C   STRING(1024)
+	)	 PRIMARY KEY (A)`})
+	if err != nil {
+		t.Error(err)
```

Comment:
nit: This should be `Fatal` as the tests cannot run without the table. Same for the below DML.

Reply:
fixed